### PR TITLE
partition_sms: replace SMS MVs with TMVP MVs, retrain weights

### PR DESCRIPTION
**Pull request by YixinDu1**
Update the SMS partition pre-screener to use TMVP-projected MVs instead of motion-search MVs, and retrain the MLP on the updated feature set. Also enables reduce_prec=1 to round MLP logits to 9-bit precision, fixing non-deterministic encode results across platforms.

Anchor: b42be4ef
Test condition: CTC, RA, 33 frames, speed 4

RA:
```
+------------------+-------+-------+-------+-------+-------+
| Class            |     Y |    Cb |    Cr |  wAvg |  Enc% |
+------------------+-------+-------+-------+-------+-------+
| A1               |  0.20 |  0.05 |  0.14 |  0.19 |  95.4 |
| A2               |  0.13 |  0.01 |  0.18 |  0.13 |  95.1 |
| A3               |  0.20 | -0.47 | -0.13 |  0.16 |  95.4 |
+------------------+-------+-------+-------+-------+-------+
| Average w/o B2   |  0.16 | -0.05 |  0.14 |  0.15 |  96.3 |
+------------------+-------+-------+-------+-------+-------+
```

STATS_CHANGED

### CODE REVIEW

**Comment by YixinDu1** [comment]
Hi @yeqing-wu, @urvangjoshi, this patch fixes the reproducibility issue. Please find the summary below: 

image

File: av2/encoder/partition_sms.c [hunk 1]
```diff
@@ -44,11 +44,46 @@ static int sms_bsize_slot(BLOCK_SIZE bsize) {
...
  * ------------------------------------------------------------------- */
 
+/* Average TMVP MVs over a rectangular region (mi_row, mi_col, mi_h×mi_w).
+ * Returns zero MV when TMVP is unavailable or all sampled entries are invalid.
+ * TMVP is stored at 8×8-pixel (TMVP_SHIFT_BITS MI-unit) granularity; the loop
+ * steps by that stride so each entry is sampled at most once.  For sub-blocks
+ * smaller than the TMVP granularity (e.g. 4×4 quadrants of an 8×8 block, where
+ * mi_h = mi_w = 1 < step = 2) the loop still executes once, reading the TMVP
+ * entry at the top-left corner of the sub-block. */
+MV av2_get_tmvp_mv_avg(const AV2_COMMON *cm, int mi_row, int mi_col, int mi_h,
```
Comment (urvangjoshi) [inline av2/encoder/partition_sms.c]: This function seems to be defined and used in this file only, right?  If so, declare it static here (and remove from .h file).  And remove the `av2_` prefix.

File: av2/encoder/partition_sms.c [hunk 2]
```diff
@@ -90,13 +123,27 @@ static void extract_sms_features(const AV2_COMP *cpi, const MACROBLOCK *x,
...
+  const int blk_h = mi_size_high[bsize];
+  const int blk_w = mi_size_wide[bsize];
+  const int half_h = blk_h / 2;
+  const int half_w = blk_w / 2;
+  const int qr[SMS_N_SPLIT_PARTS] = { mi_row, mi_row, mi_row + half_h,
+                                      mi_row + half_h };
+  const int qc[SMS_N_SPLIT_PARTS] = { mi_col, mi_col + half_w, mi_col,
+                                      mi_col + half_w };
+
+  const MV tmvp_whole = av2_get_tmvp_mv_avg(cm, mi_row, mi_col, blk_h, blk_w);
```
Comment (urvangjoshi) [inline av2/encoder/partition_sms.c]: What is the reason to change to TMVP MVs, instead of tweaking the existing method with reduce_prec = 1? 
What's the overall loss from this method now (earlier PR + this one)?  Just want to make sure the loss is not same / bigger than completely disabling some partition types.

Comment (YixinDu1) [reply av2/encoder/partition_sms.c]: We found that the model trained with TMVP mv is a bit more accurate than simple motion search mv. This allows us to tune up the model to bring more speed ups.

Just got the full overall results of this method:

image

File: av2/encoder/partition_sms.c (hunk 1 above)
Comment (YixinDu1) [reply av2/encoder/partition_sms.c]: Fixed

File: av2/encoder/partition_sms.c (hunk 2 above)
Comment (urvangjoshi) [reply av2/encoder/partition_sms.c]: Loss on A1 is higher but ratio seems within the threshold.  I'll let @leolzhao @yunqingwang1 confirm.

File: av2/encoder/partition_sms.h [hunk 3]
```diff
@@ -10,6 +10,7 @@
 #include "av2/encoder/encoder.h"
 #include "av2/encoder/encodeframe_utils.h"
 #include "av2/encoder/partition_strategy.h"
+#include "av2/common/mv.h"
```
Comment (urvangjoshi) [inline av2/encoder/partition_sms.h]: This include should go in the `.c` file. I believe it's not needed here.

Comment (YixinDu1) [reply av2/encoder/partition_sms.h]: Fixed

File: av2/encoder/partition_sms.c (hunk 2 above)
Comment (leolzhao) [reply av2/encoder/partition_sms.c]: The ratio is pretty good, so I'm good to merge it.

**Review by leolzhao — APPROVED** [review]

**Review by urvangjoshi — APPROVED** [review]
